src/Project/transformers/createTransformerWatcher.ts

```diff
@@ -0,0 +1,62 @@
+import ts from "byots";
+import { TransformerWatcher } from "Shared/types";
+
+function createServiceHost(program: ts.Program) {
+	const rootFileNames = program.getRootFileNames().map(x => x);
+	const files = new Map<string, number>();
+
+	rootFileNames.forEach(fileName => {
+		files.set(fileName, 0);
+	});
+
+	const overridedText = new Map<string, string>();
+
+	function updateFile(fileName: string, text: string) {
+		overridedText.set(fileName, text.length ? text : "\0");
```

Comment:
Is the `\0` important here? If it is, can you add a comment explaining why?

Reply:
The \0 is necessary here, I'll add a comment explaining it

Reply:
Actually, it's not necessary here. I'll remove it.